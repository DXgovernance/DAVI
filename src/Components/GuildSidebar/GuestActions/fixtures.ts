import { GuestActionsProps } from './GuestActions';

export const withData: GuestActionsProps = {
  userWalletAddress: '0x0000000000000000000000000000000000000000',
  onShowStakeModal: jest.fn(),
  onShowWalletModal: jest.fn(),
};
