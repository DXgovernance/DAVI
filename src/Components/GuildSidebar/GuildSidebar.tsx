import dxIcon from 'assets/images/dxdao-icon.svg';
import { Loading } from 'Components/Primitives/Loading';
import { useTranslation } from 'react-i18next';
import {
  DaoBrand,
  DaoIcon,
  DaoInfo,
  DaoInfoPanel,
  DaoMemberCount,
  DaoTitle,
  SidebarMenu,
  SidebarMenuItem,
  SidebarWrapper,
} from './GuildSidebar.styled';

export interface GuildSidebarProps {
  guildName?: string;
  numberOfMembers?: number;
  actions?: React.ReactNode;
}

export const GuildSidebar: React.FC<GuildSidebarProps> = ({
  guildName,
  numberOfMembers,
  actions,
}) => {
  const { t } = useTranslation();

  return (
    <SidebarWrapper>
      <DaoInfoPanel>
        <DaoInfo>
          <DaoBrand>
            <DaoIcon src={dxIcon} alt={guildName} />

            <DaoTitle size={2} as="h1">
              {guildName || <Loading loading text />}
            </DaoTitle>
          </DaoBrand>
          <DaoMemberCount>
            {t('members', { count: numberOfMembers })}
          </DaoMemberCount>
        </DaoInfo>
        {actions}
      </DaoInfoPanel>
      <SidebarMenu>
        <SidebarMenuItem href="#">{t('proposals')}</SidebarMenuItem>
        <SidebarMenuItem href="#">{t('members')}</SidebarMenuItem>
        <SidebarMenuItem href="#">{t('portfolio')}</SidebarMenuItem>
        <SidebarMenuItem href="#">{t('settings')}</SidebarMenuItem>
      </SidebarMenu>
    </SidebarWrapper>
  );
};
