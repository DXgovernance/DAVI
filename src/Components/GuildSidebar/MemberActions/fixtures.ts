import { BigNumber } from 'ethers';
import moment from 'moment';
import { MemberActionsProps } from './MemberActions';

export const propsWithData: MemberActionsProps = {
  guildToken: {
    decimals: 18,
    name: 'Guild Token',
    symbol: 'GT',
    totalSupply: BigNumber.from('1000000000'),
  },
  userWalletAddress: '0x1234567890123456789012345678901234567890',
  userEnsAvatar: {
    imageUrl: '/assets/favicon-32x32.png',
    ensName: 'test.eth',
  },
  isRepGuild: false,
  unlockedAt: moment().add(1, 'day'),
  userVotingPower: BigNumber.from('250000000'),
  userVotingPowerPercent: 50,
};
