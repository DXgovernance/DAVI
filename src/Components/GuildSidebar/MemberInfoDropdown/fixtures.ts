import { BigNumber } from 'ethers';
import moment from 'moment';
import { MemberInfoDropdownProps } from './MemberInfoDropdown';

export const closedWithoutData: MemberInfoDropdownProps = {
  isOpen: false,
  onClose: jest.fn(),
};

export const openWithoutData: MemberInfoDropdownProps = {
  isOpen: true,
  onClose: jest.fn(),
};

export const closedWithData: MemberInfoDropdownProps = {
  isOpen: false,
  onClose: jest.fn(),
  guildToken: {
    decimals: 18,
    name: 'Guild Token',
    symbol: 'GT',
    totalSupply: BigNumber.from('1000000000'),
  },
  isRepGuild: false,
  unlockedAt: moment().add(1, 'day'),
  userVotingPower: BigNumber.from('250000000'),
  userVotingPowerPercent: 50,
  onShowStakeModal: jest.fn(),
  onWithdraw: jest.fn(),
};

export const openWithData: MemberInfoDropdownProps = {
  ...closedWithData,
  isOpen: true,
};
