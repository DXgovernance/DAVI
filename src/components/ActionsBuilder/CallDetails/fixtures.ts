import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
