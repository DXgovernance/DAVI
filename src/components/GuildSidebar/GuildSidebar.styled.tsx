import styled from 'styled-components';
import { Box } from 'components/primitives/Layout/Box';
import { Menu } from 'components/Menu';
import { Heading } from 'components/primitives/Typography';

export const SidebarWrapper = styled(Box)`
  color: ${({ theme }) => theme.colors.text};
  @media only screen and (min-width: 768px) {
    margin-right: 1rem;
    border: 1px solid ${({ theme }) => theme.colors.border1};
    border-radius: ${({ theme }) => theme.radii.curved};
  }
`;

export const DaoInfoPanel = styled(Box)`
  border-bottom: 1px solid ${({ theme }) => theme.colors.border1};
  display: flex;
  flex-direction: column;
  padding: 2rem;
`;

export const DaoInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
`;

export const DaoBrand = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  @media only screen and (min-width: 768px) {
    flex-direction: column;
  }
`;

export const DaoIcon = styled.img`
  height: 3rem;
  width: 3rem;
`;

export const DaoTitle = styled(Heading)`
  margin-left: 4px;
  line-height: 1;
`;

export const DaoMemberCount = styled(Box)`
  margin-bottom: 0.5rem;
`;

export const SidebarMenu = styled(Menu)`
  margin: 0;
  padding: 1rem 0;
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  @media only screen and (min-width: 768px) {
    flex-direction: column;
  }
`;

export const SidebarMenuItem = styled.div<{ current?: boolean }>`
  padding: 0.8rem 1rem;
  color: ${({ current, theme }) =>
    current ? theme.colors.text : theme.colors.grey};

  @media only screen and (min-width: 768px) {
    border-left: 2px solid
      ${({ current, theme }) => (current ? theme.colors.text : 'transparent')};
  }

  &:hover {
    border-bottom: 2px solid ${({ theme }) => theme.colors.border1};

    @media only screen and (min-width: 768px) {
      border-left: 2px solid
        ${({ current, theme }) =>
          current ? theme.colors.text : theme.colors.border1};
      border-bottom: initial;
    }
  }

  &:active {
    border-bottom: 2px solid ${({ theme }) => theme.colors.border1};

    @media only screen and (min-width: 768px) {
      border-left: 2px solid ${({ theme }) => theme.colors.text};
      border-bottom: initial;
    }
  }
`;
