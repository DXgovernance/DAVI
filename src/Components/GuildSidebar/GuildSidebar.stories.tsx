import { ComponentStory, ComponentMeta } from '@storybook/react';
import { GuildSidebar } from './GuildSidebar';
import { withData } from './fixtures';

export default {
  title: 'Sidebars/GuildSidebar/GuildSidebar',
  component: GuildSidebar,
} as ComponentMeta<typeof GuildSidebar>;

const Template: ComponentStory<typeof GuildSidebar> = args => (
  <GuildSidebar {...args} />
);

export const Simple = Template.bind({});
Simple.args = withData;

export const Loading = Template.bind({});
Loading.args = {};
