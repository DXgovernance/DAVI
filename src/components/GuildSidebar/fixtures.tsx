import { GuildSidebarProps } from './GuildSidebar';

const TestDiv = () => <div>Test</div>;

export const withData: GuildSidebarProps = {
  guildName: 'Test Guild',
  numberOfMembers: 10,
  actions: TestDiv(),
};
