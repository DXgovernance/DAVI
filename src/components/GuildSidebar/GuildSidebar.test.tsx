import { render } from '../../utils/tests';
import '@testing-library/jest-dom';
import { GuildSidebar } from './GuildSidebar';
import { withData } from './fixtures';

jest.mock('Modules/Guilds/Hooks/useTypedParams', () => ({
  useTypedParams: () => ({
    chainName: 'localhost',
    guildId: '0xE9bDaB08f2FBb370d2a6F6661a92d9B6157E9fd2',
  }),
}));

describe('GuildSidebar', () => {
  it(`Should render without data`, async () => {
    const { container } = render(<GuildSidebar />);
    expect(container).toMatchSnapshot();
  });

  it(`Should render with data`, async () => {
    const { container } = render(<GuildSidebar {...withData} />);
    expect(container).toMatchSnapshot();
  });
});
