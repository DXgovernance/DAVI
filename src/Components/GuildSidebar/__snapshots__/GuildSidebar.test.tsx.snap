// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuildSidebar Should render with data 1`] = `
<div>
   
  <div
    class="sc-bdfBQB sc-iBPTik bmhbKs usvUU"
  >
    <div
      class="sc-bdfBQB sc-fubCzh bmhbKs hNwxkY"
    >
      <div
        class="sc-pGacB coA-Dha"
      >
        <div
          class="sc-jrAFXE jlptGR"
        >
          <img
            alt="DXdao Logo"
            class="sc-kEjbQP gWbxpV"
            src="dxdao-icon.svg"
          />
          <h1
            class="sc-gKseQn sc-iqHYmW fMfHRo iyXQFt"
            size="2"
          >
            Test Guild
          </h1>
        </div>
        <div
          class="sc-bdfBQB sc-crrszt bmhbKs uTbjj"
        >
          10
           Members
        </div>
      </div>
    </div>
    <div
      class="sc-bdfBQB sc-eCstlR sc-dQoVA bmhbKs hPdNYY fnEvkC"
    >
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Proposals
      </a>
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Members
      </a>
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Portfolio
      </a>
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Settings
      </a>
    </div>
  </div>
   
</div>
`;

exports[`GuildSidebar Should render without data 1`] = `
<div>
   
  <div
    class="sc-bdfBQB sc-iBPTik bmhbKs usvUU"
  >
    <div
      class="sc-bdfBQB sc-fubCzh bmhbKs hNwxkY"
    >
      <div
        class="sc-pGacB coA-Dha"
      >
        <div
          class="sc-jrAFXE jlptGR"
        >
          <img
            alt="DXdao Logo"
            class="sc-kEjbQP gWbxpV"
            src="dxdao-icon.svg"
          />
          <h1
            class="sc-gKseQn sc-iqHYmW fMfHRo iyXQFt"
            size="2"
          />
        </div>
        <div
          class="sc-bdfBQB sc-crrszt bmhbKs uTbjj"
        >
           Members
        </div>
      </div>
    </div>
    <div
      class="sc-bdfBQB sc-eCstlR sc-dQoVA bmhbKs hPdNYY fnEvkC"
    >
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Proposals
      </a>
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Members
      </a>
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Portfolio
      </a>
      <a
        class="sc-jSgvzq sc-bqyKOL RJAec kmntIG"
        href="#"
      >
        Settings
      </a>
    </div>
  </div>
   
</div>
`;
